migrations/migrations/0012_committee_user_relation.py

```diff
@@ -234,65 +228,69 @@ def handle_user_restore_event(
         return [
             ListUpdateEvent(
                 f"committee/{committee_id}",
-                {"add": {f"user_${cml_permission}_management_level": [user_id]}},
+                {"add": {committee_cml_permission_field: [user_id]}},
             )
             for committee_id in cml_committee_ids
         ]
 
     def handle_user_update_event(
         self, event: UpdateEvent, user: Dict[str, Any], user_id: int
     ) -> Optional[List[BaseEvent]]:
-        meeting_ids = list(map(int, event.data.get("group_$_ids", []))) or []
-        new_committee_ids = set(
-            self.new_accessor.get_model_ignore_deleted(f"meeting/{meeting_id}")[0].get(
-                "committee_id"
-            )
-            for meeting_id in meeting_ids
-        )
-        new_cml_committee_ids = list(
-            map(
-                int, event.data.get(f"committee_${cml_permission}_management_level", [])
-            )
-        )
-        new_committee_ids.update(new_cml_committee_ids)
         meeting_ids = list(map(int, cast(List[str], user.get("group_$_ids", [])))) or []
         old_committee_ids = set(
             self.new_accessor.get_model_ignore_deleted(f"meeting/{meeting_id}")[0].get(
                 "committee_id"
             )
             for meeting_id in meeting_ids
         )
-        old_cml_committee_ids = cast(
-            List[int], user.get(f"committee_${cml_permission}_management_level", [])
-        )
+        if "group_$_ids" in event.data:
+            meeting_ids = list(map(int, event.data["group_$_ids"]))
+            new_committee_ids = set(
+                self.new_accessor.get_model_ignore_deleted(f"meeting/{meeting_id}")[
+                    0
+                ].get("committee_id")
+                for meeting_id in meeting_ids
+            )
+        else:
+            new_committee_ids = old_committee_ids.copy()
+
+        old_cml_committee_ids = cast(List[int], user.get(user_cml_permission_field, []))
+        if user_cml_permission_field in event.data:
```

Comment:
Why should be `user_cml_permission_field` in event.data? This would only happen, if you migrate a part twice.

Reply:
It appears, because in migrate_event the event was changed from original format (committee_$1_management_level) to new format (this one, committee_$can_manage_management_level). More can't be done in migrate_event, because the necessary meeting possibly still not exists. The (modified) event and, if it exists, the user in this state, are gathered in a list self.event_user.
In get_additional_events all events are stored in "database", the necessary meeting is created. Now get_additional_events works on the list self.event_users and calls the handle_user_update_event (and others) with the same event (in new format, see above) again.